packages/core/src/tools/readBytesFromStream.ts

```diff
@@ -0,0 +1,75 @@
+import { monitor } from './monitor'
+import { noop } from './utils'
+
+type Options = {
+  limit: number
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏ ‏Maybe the naming could be more explicit like `bytesLimit`